File: codigo/python/generator.py
import pika
import random
import json
import time

connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
channel = connection.channel


class Generator:

    def __init__(self):
        self.cities = []
        self.cities.append({"name": "Aveiro", "district": "Aveiro", "lat": 40.6412, "lon": -8.65362})
        self.cities.append({"name": "Arouca", "district": "Aveiro", "lat": 40.9289, "lon": -8.24364})
        self.cities.append({"name": "Sintra", "district": "Lisboa", "lat": 38.7984, "lon": -9.38811})
        for i in range(len(self.cities)):
            self.cities[i]["id"] = i + 1

    def __generate(self):
        for city in self.cities:
            city["co"] = random.randint(0, 1200)  # bppv
            city["co2"] = random.randint(38, 420)  # ppm
            city["so2"] = random.randint(5, 35)  # micrg/m^3
            city["noise"] = random.randint(15, 45)  # db
            city["rainPh"] = random.randint(0, 7)

    def clock(self):
        while True:
            self.__generate()
            for city in self.cities:
                connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
                channel = connection.channel()
                QUEUE_NAME = str(city["id"])
                channel.queue_declare(queue = QUEUE_NAME)
                message = json.dumps(city)
                channel.basic_publish(exchange='',
                      routing_key=QUEUE_NAME,
                      body=message)
                print("[generator] sent " + message)
                connection.close()
            time.sleep(60)


def main():
    generator = Generator()
    generator.clock()


if __name__ == "__main__":
    main()

File: Projeto/cityStats/src/main/java/ua/ies/project/cityStats/Generator.java
package ua.ies.project.cityStats;

import java.nio.charset.StandardCharsets;
import java.util.*;

import com.rabbitmq.client.ConnectionFactory;
import com.rabbitmq.client.Connection;
import com.rabbitmq.client.Channel;

@Deprecated
public class Generator {

    private ArrayList<City> cities;

    public Generator(){
        cities = new ArrayList<>();
        cities.add(new City("Aveiro", "Aveiro", 40.6412, -8.65362));
        cities.add(new City("Arouca", "Aveiro", 40.9289, -8.24364));
        cities.add(new City("Sintra", "Lisboa", 38.7984, -9.38811));
        for (int i = 0; i < cities.size(); i++){
            cities.get(i).setId((long) (i + 1));
        }
    }

    private void generate(){
        Random random = new Random();
        for (City city : cities){
            city.setCo(random.nextInt(1200)); //ppbv
            city.setCo2(random.nextInt(38) + 382); //ppm
            city.setSo2(random.nextInt(30) + 5); //microg/m^3
            city.setNoise(random.nextInt(30) + 15); //db
            city.setRainPh(random.nextInt(7));
        }
    }

    public void clock() throws Exception {
        ConnectionFactory factory = new ConnectionFactory();
        factory.setHost("localhost"); //create connections to localhost
        try (Connection connection = factory.newConnection();
        Channel channel = connection.createChannel()){
            while (true) {
                generate();  //generate data for each city
                for (City city : cities){
                    String QUEUE_NAME = city.getId() + "";
                    channel.queueDeclare(QUEUE_NAME, false, false, false, null);
                    String message = city.json();
                    channel.basicPublish("", QUEUE_NAME, null, message.getBytes(StandardCharsets.UTF_8));
                    System.out.println(" [generator] sent '" + message + "'");
                }
                Thread.sleep(60000);
            }
        }
    }
}
